otcextensions/sdk/cloud/dds.py

```diff
@@ -0,0 +1,186 @@
+                )
+            attrs['mode'] = mode
+
+        flavors_ref = list(self.dds.flavors(
+            region=region,
+            engine_name=datastore_type)
+        )
+        flavors_specs = [flavor.spec_code for flavor in flavors_ref]
+        if flavors:
+            for flavor in flavors:
+                if flavor['type'] in ['mongos', 'shard'] \
+                        and flavor['num'] not in range(2, 16):
+                    raise exceptions.SDKException(
+                        '`num` value must be in ranges from 2 to 16 '
+                        'for mongos and shard'
+                    )
+                if flavor['type'] in ['config', 'replica'] \
+                        and flavor['num'] != 1:
+                    raise exceptions.SDKException(
+                        '`num` value must be 1 '
+                        'for config and replica'
+                    )
+                if flavor['type'] == 'mongos':
+                    if all(k in flavor for k in ('storage', 'size')):
+                        raise exceptions.SDKException(
+                            '`storage` and `size` parameters'
+                            ' is invalid for the mongos nodes'
+                        )
+                if 'size' in flavor:
+                    if flavor['type'] == 'replica' \
+                            and not (10 <= flavor['size'] <= 2000):
+                        raise exceptions.SDKException(
+                            '`size` value for `replica` must be'
+                            ' between 10 and 2000 GB.'
+                        )
+                    elif flavor['type'] == 'config' \
+                            and flavor['size'] != 20:
+                        raise exceptions.SDKException(
+                            '`size` value for `config` must be 20 GB.'
+                        )
+                    elif not (10 <= flavor['size'] <= 1000):
+                        raise exceptions.SDKException(
+                            '`size` value for `shard` must be'
+                            ' between 10 and 1000 GB.'
+                        )
+                if flavor['spec_code'] not in flavors_specs:
+                    raise exceptions.SDKException(
+                        '`spec_code` not valid'
+                    )
+            attrs['flavor'] = flavors
+
+        if backup_keepdays and backup_timeframe:
+            backup_attrs = {}
+            backup_attrs['keep_days'] = backup_keepdays
+            backup_attrs['start_time'] = backup_timeframe
+            attrs['backup_strategy'] = backup_attrs
+        elif backup_keepdays or backup_timeframe:
+            raise exceptions.SDKException(
+                '`backup_keepdays` and `backup_timeframe` must be passed'
+                'together'
+            )
+
+        if ssl_option:
+            attrs['ssl_option'] = ssl_option
+
+        obj = self.dds.create_instance(**attrs)
+        obj = self.dds.get_instance(obj.id)
```

Comment:
without implementing wait (should be similar to RDS, not sure though whether status querying API is available) it makes little to no sense doing get immediately after create.

Reply:
dds does not have /jobs endpoint